improving formatting of warnings/errors

I find the warnings/error output from agda slightly confusing due to lowercase "when".
To me the lowercase suggests `when checking` might be continuing from earlier output,
but it seems looking at the source code that `when checking` actually starts a warning/error message trace.

Personally I would prefer to write it as `When checking` to make it clearer that starting a warning/error (ie it is about what follows not any preceding lines).
(Of course this might be considered a breaking change since it would be changing the warning/error output format.)

* [x] My report includes the version of Agda (and Emacs, GHC, and OS versions where relevant).

agda-2.6.4 , ghc-9.4.5, Fedora Linux

* [x] The issue can be reproduced with the _latest version_ of Agda (either [released](https://wiki.portal.chalmers.se/agda/Main/Download) or `master` branch).

* [x] I have supplied all information so the bug can be easily reproduced (in particular, the reproducer is a complete Agda file).

I am getting warnings/error when building agda-stdlib-{2.7.2,2.7.3} with Agda 2.6.4 (see agda/agda-stdlib#2129 a little context)

"when checking..." is placed at the _end_ of the error message.
The current format is:
1. location, for warnings also: warning identifier
2. message
3. context ("when checking...")

So one warning of your OP prints like this:
```
/var/home/petersen/fedora/haskell/Agda-stdlib/BUILD/agda-stdlib-1.7.3/src/Algebra/Construct/LexProduct/Inner.agda:258,1-260,32
warning: -W[no]UnsupportedIndexedMatch
This clause uses pattern-matching features that are not yet
supported by Cubical Agda, the function to which it belongs will
not compute when applied to transports.

Reason: It relies on injectivity of the data constructor false,
        which is not yet supported

when checking the definition of
Algebra.Construct.LexProduct.Inner.with-1214
```

Likely this could be improved by indentation and a blank line at the end.

In general, we are aspiring to follow the format of GHC errors:
1. a. location, b. error code
2. error
3. context
5. printing the context
```
Foo.hs:1:8: error: [GHC-39999]
    • No instance for ‘Num (IO t0)’ arising from the literal ‘5’
    • In the expression: 5
      In an equation for ‘main’: main = 5
  |
1 | main = 5
  |        ^
```

--- a/src/full/Agda/Interaction/Imports.hs
+++ b/src/full/Agda/Interaction/Imports.hs
@@ -699,7 +699,7 @@ getStoredInterface x file msrc = do
         let ws = filter ((Strict.Just (Just x) ==) .
                          fmap rangeFileName . tcWarningOrigin) $
                  iWarnings i
-        unless (null ws) $ alwaysReportSDoc "warning" 1 $ P.vcat $ P.prettyTCM <$> ws
+        unless (null ws) $ alwaysReportSDoc "warning" 1 $ P.vsep $ map P.prettyTCM ws
 
         loadDecodedModule file $ ModuleInfo
           { miInterface = i
@@ -971,7 +971,7 @@ createInterface mname file isMain msrc = do
                                      fmap rangeFileName . tcWarningOrigin) $
                              tcWarnings classified
                    unless (null wa') $
-                     alwaysReportSDoc "warning" 1 $ P.vcat $ P.prettyTCM <$> wa'
+                     alwaysReportSDoc "warning" 1 $ P.vsep $ map P.prettyTCM wa'
                    when (null (nonFatalErrors classified)) $ chaseMsg "Finished" x Nothing)
 
   withMsgs $

--- a/src/full/Agda/Main.hs
+++ b/src/full/Agda/Main.hs
@@ -330,7 +330,7 @@ runTCMPrettyErrors tcm = do
           `catchError` \err -> do
             s2s <- prettyTCWarnings' =<< getAllWarningsOfTCErr err
             s1  <- prettyError err
-            ANSI.putDoc (P.vcat s2s P.$+$ s1)
+            ANSI.putDoc $ P.vsep $ s2s ++ [ s1 ]
             liftIO $ do
               helpForLocaleError err
             return (Just TCMError)

--- a/src/full/Agda/Syntax/Common/Pretty.hs
+++ b/src/full/Agda/Syntax/Common/Pretty.hs
@@ -10,6 +10,7 @@ module Agda.Syntax.Common.Pretty
 
 import Prelude hiding (null)
 
+import qualified Data.List as List
 import qualified Data.Foldable as Fold
 import qualified Data.IntSet as IntSet
 import qualified Data.IntMap as IntMap
@@ -182,6 +183,9 @@ punctuate :: Foldable t => Doc -> t Doc -> [Doc]
 punctuate d = P.punctuate d . Fold.toList
 
 -- * 'Doc' utilities
+
+vsep :: [Doc] -> Doc
+vsep = vcat . List.intersperse ""
 
 pwords :: String -> [Doc]
 pwords = map text . words

--- a/src/full/Agda/TypeChecking/Pretty.hs
+++ b/src/full/Agda/TypeChecking/Pretty.hs
@@ -97,12 +97,13 @@ pwords s = map pure $ P.pwords s
 fwords :: Applicative m => String -> m Doc
 fwords s = pure $ P.fwords s
 
-sep, fsep, hsep, hcat, vcat :: (Applicative m, Foldable t) => t (m Doc) -> m Doc
+sep, fsep, hsep, hcat, vcat, vsep :: (Applicative m, Foldable t) => t (m Doc) -> m Doc
 sep ds  = P.sep  <$> sequenceA (Fold.toList ds)
 fsep ds = P.fsep <$> sequenceA (Fold.toList ds)
 hsep ds = P.hsep <$> sequenceA (Fold.toList ds)
 hcat ds = P.hcat <$> sequenceA (Fold.toList ds)
 vcat ds = P.vcat <$> sequenceA (Fold.toList ds)
+vsep ds = P.vsep <$> sequenceA (Fold.toList ds)
 
 hang :: Applicative m => m Doc -> Int -> m Doc -> m Doc
 hang p n q = P.hang <$> p <*> pure n <*> q

